libvast/vast/concept/parseable/core/choice.hpp

```diff
@@ -83,26 +83,20 @@ class choice_parser : public parser<choice_parser<Lhs, Rhs>> {
   }
 
 private:
-  template <class Left, class Iterator, class Attribute>
-  auto parse_left(Iterator& f, const Iterator& l, Attribute& a) const
-  -> std::enable_if_t<is_choice_parser<Left>{}, bool> {
-    return lhs_(f, l, a); // recurse
-  }
-
-  template <class Left, class Iterator>
-  auto parse_left(Iterator& f, const Iterator& l, unused_type) const
-  -> std::enable_if_t<!is_choice_parser_v<Left>, bool> {
-    return lhs_(f, l, unused);
-  }
-
-  template <class Left, class Iterator, class Attribute>
-  auto parse_left(Iterator& f, const Iterator& l, Attribute& a) const
-  -> std::enable_if_t<!is_choice_parser_v<Left>, bool> {
-    lhs_attribute al;
-    if (!lhs_(f, l, al))
-      return false;
-    a = std::move(al);
-    return true;
+  template <class Iterator, class Attribute>
+  bool parse_left(Iterator& f, const Iterator& l, Attribute& a) const {
+    using detail::is_any_v;
+    if constexpr (is_choice_parser_v<Lhs>) {
+      return lhs_(f, l, a); // recurse
```

Comment:
I assume this is an optimization to avoid the temporary `lhs_attribute` on line 94? Could this be done for `parse_right` as well?

Reply:
Yep.